cmake/HPX_AddModule.cmake

```diff
@@ -61,33 +61,14 @@ function(add_hpx_module libname modulename)
             FORCE
   )
 
-  # HPX options
-  if(${_have_compatibility_headers_option})
-    set(_compatibility_headers_default OFF)
-    if(${modulename}_COMPATIBILITY_HEADERS)
-      set(_compatibility_headers_default ON)
-    endif()
-    hpx_option(
-      HPX_${modulename_upper}_WITH_COMPATIBILITY_HEADERS
-      BOOL
-      "Enable compatibility headers for old headers. (default: ${_compatibility_headers_default})"
-      ${_compatibility_headers_default}
-      ADVANCED
-      CATEGORY "Modules"
-      MODULE ${modulename_upper}
-    )
-  endif()
-
   # Main directories of the module
   set(SOURCE_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/src")
   set(HEADER_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/include")
 
   hpx_debug("Add module ${modulename}: SOURCE_ROOT: ${SOURCE_ROOT}")
   hpx_debug("Add module ${modulename}: HEADER_ROOT: ${HEADER_ROOT}")
 
-  if(${_have_compatibility_headers_option}
-     AND HPX_${modulename_upper}_WITH_COMPATIBILITY_HEADERS
-  )
+  if(${_have_compatibility_headers})
```

Comment:
```suggestion
  if(${modulename}_COMPAT_HEADERS)
```
?
Same below.

Reply:
If some compatibility headers are specified in the CMakeLists.txt of the module, it would install the compatibility headers etc., if none are specified then nothing is done. Did I miss something?
It was to match the modules like the config ones where there is no compatibility headers.

Reply:
The behaviour is correct, I was just proposing an alternative, but it might be subjective if it's clearer. Instead of
```
if(${modulename}_COMPAT_HEADERS)
  set(_have_compatibility_headers TRUE)
else()
  set(_have_compatibility_headers FALSE)
endif()

# later
if(${_have_compatibility_headers})
  # do something if compatibility headers are there
endif()
```
I would just do 
```
# later
if(${modulename}_COMPAT_HEADERS)
  # do something if compatibility headers are there
endif()
```
Please keep whichever one you find clearer though. I might be the odd one on this.

Reply:
Sure initially we just did that to keep the install steps together, etc. but we can totally do that!

Reply:
Note that the install steps can still be together. I just mean that the extra `_have_compatibility_headers` variable is unnecessary (to me).

Reply:
Yep okay I don't have any strong preference, but you're right that it's probably clearer. Sorry for the confusion